### PR TITLE
[sailfish-webview] Make PermissionModel::setPermissionList to handle insert and remove. Fixes JB#52375

### PR DESCRIPTION
**Pull request by atatarov**

### CODE REVIEW

**Review by rainemak — CHANGES_REQUESTED** [review]
Not like this.

You should compare permission list parameter and current status of the m_permissionList and do row removes and inserts accordingly.

**Comment by atatarov** [comment]
testing needed

**Review by llewelld — COMMENTED** [review]
The code for updating the model looks sensible to me. I didn't notice any issues when testing it either.

Nice changes. I've made just one suggestion.

File: import/controls/permissionmodel.cpp [hunk 1]
```diff
@@ -139,19 +134,46 @@ void PermissionModel::handleRecvObserve(const QString &message, const QVariant &
...
+                startIndex = i;
+            }
+        } else if (startIndex >= 0) {
+            emit dataChanged(index(startIndex), index(i-1));
+            startIndex = -1;
+        }
+    }
+
+    if (startIndex >= 0) {
+        emit dataChanged(index(startIndex), index(m_permissionList.count()-1));
```
[the remark this answers is not in the document]
Comment (atatarov) [reply import/controls/permissionmodel.cpp]: Thank you, done

**Review by llewelld — APPROVED** [review]
This looks good to me, and also seemed to work fine during brief testing.

**Review by okodron — APPROVED** [review]

**Review by rainemak — APPROVED** [review]
@llewelld already approved. Approving as well.

**Pull request merged**